src/cli.ts

```diff
@@ -25,26 +40,50 @@ async function main(argv: minimist.ParsedArgs) {
   }
 
   const argIn: string = argv._[0] || argv.input
-  const argOut: string = argv._[1] || argv.output
+  const argOut: string | undefined = argv._[1] || argv.output
 
   try {
-    const schema: JSONSchema4 = JSON.parse(await readInput(argIn))
-    const ts = await compile(schema, argIn, argv as Partial<Options>)
-    await writeOutput(ts, argOut)
+    await processFiles(argIn, argOut, argv as Partial<Options>)
   } catch (e) {
     console.error(whiteBright.bgRedBright('error'), e)
     process.exit(1)
   }
 }
 
+async function processFiles(argIn: string, argOut: string | undefined, argv: Partial<Options>): Promise<void[]> {
+  const files = isGlob(argIn) ? await glob(join(process.cwd(), argIn)) : [argIn]
+
+  if (files.length === 0) {
+    throw ReferenceError(
+      `You passed a glob pattern "${argIn}", but there are no files that match that pattern in ${process.cwd()}`
+    )
+  }
+
+  if (argOut && extname(argOut) === '' && !existsSync(argOut)) {
+    await mkdirp(argOut)
+  }
+
+  return Promise.all(files.map(file => processFile(file, argOut, argv)))
+}
+
+async function processFile(file: string, out: string | undefined, argv: Partial<Options>): Promise<void> {
+  out = out ? join(process.cwd(), out) : ''
```

Comment:
What does it mean for `out` to be `''` here? What will the behavior be (and could we make that behavior more explicit?

Also:

- It might be worth killing `writeOutput`, and splitting it into `writeOutputToStdOut` and a regular `writeFile` call
- It would be nice to avoid querying the filesystem for whether `out` is a directory or not. Instead, we should always pass a file path to `processFile`.